src/uikit.cs

```diff
@@ -16610,6 +16754,505 @@ interface UITargetedDragPreview : NSCopying
+
+		[Export ("initWithActivationHandler:")]
+		IntPtr Constructor (Action<UISpringLoadedInteraction, IUISpringLoadedInteractionContext> handler);
+
+		[Export ("interactionBehavior", ArgumentSemantic.Strong)]
+		IUISpringLoadedInteractionBehavior InteractionBehavior { get; }
+
+		[Export ("interactionEffect", ArgumentSemantic.Strong)]
+		IUISpringLoadedInteractionEffect InteractionEffect { get; }
+	}
+
+	interface IUISpringLoadedInteractionBehavior {}
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	[Protocol]
+	interface UISpringLoadedInteractionBehavior {
+		[Abstract]
+		[Export ("shouldAllowInteraction:withContext:")]
+		bool ShouldAllowInteraction (UISpringLoadedInteraction interaction, IUISpringLoadedInteractionContext context);
+
+		[Export ("interactionDidFinish:")]
+		void InteractionDidFinish (UISpringLoadedInteraction interaction);
+	}
+
+	interface IUISpringLoadedInteractionEffect {}
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	[Protocol]
+	interface UISpringLoadedInteractionEffect {
+		[Abstract]
+		[Export ("interaction:didChangeWithContext:")]
+		void DidChange (UISpringLoadedInteraction interaction, IUISpringLoadedInteractionContext context);
+	}
+
+	interface IUISpringLoadedInteractionContext {}
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	[Protocol]
+	interface UISpringLoadedInteractionContext {
+		[Abstract]
+		[Export ("state")]
+		UISpringLoadedInteractionEffectState State { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("targetView", ArgumentSemantic.Strong)]
+		UIView TargetView { get; set; }
+
+		[Abstract]
+		[NullAllowed, Export ("targetItem", ArgumentSemantic.Strong)]
+		NSObject TargetItem { get; set; }
+
+		[Abstract]
+		[Export ("locationInView:")]
+		CGPoint LocationInView ([NullAllowed] UIView view);
+	}
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	[Protocol]
+	interface UISpringLoadedInteractionSupporting {
+		[Abstract]
+		[Export ("springLoaded")]
+		bool SpringLoaded { [Bind ("isSpringLoaded")] get; set; }
+	}
+
+	// https://bugzilla.xamarin.com/show_bug.cgi?id=58282, we should be able to write one delegate with a 'Action<bool>'. See original signature:
+	// typedef void (^UIContextualActionHandler)(UIContextualAction * _Nonnull, __kindof UIView * _Nonnull, void (^ _Nonnull)(BOOL));
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	delegate void UIContextualActionHandler (UIContextualAction action, UIView sourceView, [BlockCallback] UIContextualActionCompletionHandler completionHandler);
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	delegate void UIContextualActionCompletionHandler (bool finished);
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	interface UIContextualAction {
+		[Static]
+		[Export ("contextualActionWithStyle:title:handler:")]
+		UIContextualAction FromContextualActionStyle (UIContextualActionStyle style, [NullAllowed] string title, UIContextualActionHandler handler);
```

Comment:
this static method makes me think `init` should not be allowed -> `[DisableDefaultCtor]`